test_build_tag_push.py

```diff
@@ -0,0 +1,86 @@
+import unittest
+import build_tag_push
+import os
+import subprocess
+import argparse
+
+# args_app_yml_path = '/Users/alyssakelley/Desktop/test_mvp/ymltest.yml'
+# args_docker_hub_name = 'alyssakelley'
+
+# Question: as we talked about the parses before, I then tried to put them
+# in there own function and use them later on, but I still had to use the
+# special unittest.main line below. Should I take all of this away and
+# then just use hard coded test cases (like the path and docker_hub_name
+# as I have commented out above)?
+
+
+parser = argparse.ArgumentParser(description='Usage: python3'
+                                 'build_tag_push.py <docker_hub_name'
+                                 '<app_yml_path')
+parser.add_argument('docker_hub_name', type=str, help='DockerHub namespace')
+
+parser.add_argument('app_yml_path', type=str,
+                    help='Repository path to MVP Studio file')
+args = parser.parse_args()
+
+
+class TestBuildTagPush(unittest.TestCase):
+
+    def test_parse_app_file(self):  # Success.
+
+        data_from_app_yml = build_tag_push.parse_app_file(args.app_yml_path)
+        self.assertEqual(data_from_app_yml['name'], 'build-setup-test')
+        self.assertEqual(data_from_app_yml['build'], 'run_my_test_build.sh')
+        self.assertEqual(data_from_app_yml['test'], 'test_the_test.sh')
+        self.assertEqual(data_from_app_yml['docker_context'],
+                         ['build', 'static', 'Dockerfile'])
+
+    def test_build_check_app(self):  # Fails.
+
+        data_from_app_yml = build_tag_push.parse_app_file(args.app_yml_path)
+
+        result = subprocess.check_call(['bash', data_from_app_yml['test']])
```

Comment:
I think this is failing because **check**_call throws if the run didn't succeed.

You need this here because you're taking the name/path to the yml from command line arguments so you can't control what you get. On the other hand, using the directory structure proposed above you can (and should) have **two** tests like:

```
def test_build_check_app_throws_on_build_fail(self):
    data_from_app_yml = build_tag_push.parse_app_file('test/data/build_fails_example/app.yml')
    with self.assertRaises(BuildFailedException) as shouldThrow:
        build_tag_push.build_check_app(data_from_app_yml)

def test_build_check_app_runs_build(self):
    # here I assume the build.sh in build_succeeds_example puts a file in /tmp so you can check that
    # it was actually run - see the final assert below
    os.unlink('/tmp/fake_build_output')
    data_from_app_yml = build_tag_push.parse_app_file('test/data/build_succeeds_example/app.yml')
    build_tag_push.build_check_app(data_from_app_yml)
    self.assertTrue(os.path.exists('/tmp/fake_build_output'))
```